gene/compare.go

```diff
@@ -0,0 +1,180 @@
+package gene
+
+import (
+	"errors"
+	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
+)
+
+func equal(alpha, beta *Gene) (bool, error) {
+	if alpha.id != beta.id {
+		return false, errors.New("id")
```

Comment:
This was really just intended as a testing function that will return the difference. Not sure what the utility of an equal function would be when actually using gene. In most cases I imagine it would be more useful just to compare the part you were interested in.